openlcs/libs/corgi.py

```diff
@@ -345,6 +346,23 @@ def get_product_version(self, name, fields=None) -> dict:
                 retval[field] = product_data[field]
         return retval
 
+    def deduplicate_provides(self, oci_noarch_provides):
+        # Deduplicate provides for components with different arch.
+        nvrs = []
+        provides_rs = []
+        provides_rpm = []
+        for oci_noarch_provide in oci_noarch_provides:
+            purl = oci_noarch_provide.get('purl')
+            purl_dict = PackageURL.from_string(purl).to_dict()
+            if purl_dict.get('namespace') == 'redhat':
```

Comment:
purl of rpm components starts with the 'pkg:rpm'.

Reply:
Updated.